examples/full_example/main.py

```diff
@@ -143,30 +108,45 @@ def __init__(self, **kwargs):
         change_statusbar_color(self.theme_cls.primary_color)
 
     def build(self):
-        self.mainkv = Builder.load_string(kv)
-        return self.mainkv
+        self.root = Builder.load_string(kv)
 
     def on_start(self):
-        for screen in self.screens:
-            self.mainkv.ids.sm.add_widget(eval("Factory.%s()" % screen))
-
-        for list_item in self.screens:
-            self.mainkv.ids.menu_list.add_widget(
-                Factory.MyMenuItem(
-                    text=list_item, on_release=self.list_menu_callback
+        with open("screens.json") as read_file:
+            self.data_screens = ast.literal_eval(read_file.read())
+            data_screens = list(self.data_screens.keys())
+            data_screens.sort()
+
+        for list_item in data_screens:
+            self.root.ids.menu_list.add_widget(
+                IconListItem(
+                    text=list_item,
+                    icon=self.data_screens[list_item]["icon"],
+                    on_release=lambda x=list_item: self.set_screen(x),
                 )
             )
 
-    def list_menu_callback(self, instance):
-        self.show_screen(instance.text)
-        self.mainkv.ids.navdrawer.set_state("close")
+    def set_screen(self, screen_name):
+        manager = self.root.ids.screen_manager
+        screen_details = self.data_screens[screen_name.text]
+
+        if not manager.has_screen(screen_details["screen_name"]):
+            exec(screen_details["import"])
+            screen_object = eval(screen_details["factory"])
```

Comment:
You could just write the class name inside json file instead of whole factory command:

`eval("Factory.%s"%screen_details["factory"])`